hudi-common/src/main/java/org/apache/hudi/common/table/read/InputSplit.java

```diff
@@ -37,27 +39,25 @@
 public class InputSplit {
   private final Option<HoodieBaseFile> baseFileOption;
   private final List<HoodieLogFile> logFiles;
+  private final Option<Iterator<Map.Entry<Serializable, BufferedRecord>>> inputs;
   private final String partitionPath;
   // Byte offset to start reading from the base file
   private final long start;
   // Length of bytes to read from the base file
   private final long length;
 
-  InputSplit(Option<HoodieBaseFile> baseFileOption, Stream<HoodieLogFile> logFiles, String partitionPath, long start, long length) {
+  InputSplit(Option<HoodieBaseFile> baseFileOption, Stream<HoodieLogFile> logFiles, String partitionPath, long start, long length,
```

Comment:
Provide two constructors: one taking `baseFileOption` and `logFiles`, another taking `baseFileOption` and `recordIterator`.

Reply:
I am taking in Either as an arg. 
`Either<Stream<HoodieLogFile>,Iterator<Pair<Serializable, BufferedRecord>>> logFilesOrRecordIterator`

Reply:
Usually the caller should not use `Either` when constructing the instance to reduce complexity, and rely on builder to pass in necessary components (see the builder example here: https://github.com/apache/hudi/blob/f51c36fd64398718c039b26c15767e793b82904e/hudi-common/src/main/java/org/apache/hudi/io/storage/HFileReaderFactory.java#L89).  Since there is no builder here, it's ok for now.